src/query/ts/m3db/encoded_step_iterator_test.go

```diff
@@ -509,15 +510,47 @@ func setupBlock(b *testing.B, iterations int, t iterType) (block.Block, reset) {
 	}
 
 	usePools := t == stepParallel
+
+	opts := newTestOptions()
+	if usePools {
+		poolOpts := xsync.NewPooledWorkerPoolOptions()
+		readWorkerPools, err := xsync.NewPooledWorkerPool(1024, poolOpts)
+		require.NoError(b, err)
+		readWorkerPools.Init()
+		opts = opts.SetReadWorkerPool(readWorkerPools)
+	}
+
+	for _, reset := range itersReset {
+		reset()
+	}
+
+	block, err := NewEncodedBlock(iters, models.Bounds{
+		Start:    start,
+		StepSize: stepSize,
+		Duration: window,
+	}, false, block.NewResultMetadata(), opts)
+
+	require.NoError(b, err)
+	return block, func() {
+			for _, reset := range itersReset {
+				reset()
+			}
+		},
+		setupProf(usePools, iterations)
+}
+
+func setupProf(usePools bool, iterations int) stop {
+	var pCPU, pMem interface {
```

Comment:
nit: I think functionally they're exclusive i.e. you can only do cpu or you can only do mem; can probably collapse as a single interface? Entirely up to you though

Reply:
Oh nice I didn't realize it was one or the other. I'll consolidate.